hub: Add initial support for resalloc

siteshwar opened the pull request.

lzaoral committed:
hub: remove dead code in waiving/service.py

... as detected by Coverity.

Fixes: c8de0ee6eb2eca1c459220ec77a393b3d4dab36f ("covscanhub/waiving: do not show differences against first scan")

lzaoral committed:
hub: remove unused functions in waiving/service.py

lzaoral committed:
hub: remove the get_defects_diff_display_by_rg function

... from waiving/service.py.

After recent changes, this function always returns the response object
unchanged if it is not `None` which is the case for the only line where it
was used.  Thus, it can be safely removed.

lzaoral committed:
hub: remove the get_defects_diff_display function

... from waiving/service.py.

The only place where it was called did not provide the response kwarg.
Thus, the call was equivalent to an empty dictionary.

lzaoral committed:
hub: remove dead code from templates/waiving/result.html

The `diff_state` and `diff_count` values were never set because the
corresponding code was dead.

antisaling committed:
hub: improve secret handling

Closes: https://gitlab.cee.redhat.com/covscan/covscan/-/issues/241
Signed-off-by: Isabella do Amaral <idoamara@redhat.com>

lzaoral committed:
ci: remove redundant installation of git

Which is installed by default on GitHub's machines.

lzaoral committed:
ci: do not install coverage on the host

All coverage related actions are now done in the containers so this step
is redundant.

lzaoral committed:
containers: move coverage environment setup to the corresponding script

This also fixes incompatibility with BSD sed on macOS because the -i
option is not portable.

lzaoral committed:
containers: automatically grant CREATEDB when initialising the database

... so that the developers can execute unit tests more easily.

lzaoral committed:
ci: remove commented out or unnecessary code

lzaoral committed:
ci: use the latest official Fedora container for the Lint workflow

lzaoral committed:
ci: add blank lines and comments

... to make the CI configuration easier to read.

lzaoral committed:
ci: show osh-worker logs in Django unit tests workflow

lzaoral committed:
ci: use actions/checkout to clone kobo git repository

lzaoral committed:
hub: raise 404 for URLs with unknown IDs

Resolves: https://issues.redhat.com/browse/OSH-175

lzaoral committed:
hub: rename release_list to release_stats_list

... to highlight that the method renders statistics for a given release and
not a list of releases.

lzaoral committed:
hub: raise 404 on unknown order_by values

Resolves: https://issues.redhat.com/browse/OSH-175

siteshwar committed:
hub: Add initial support for resalloc

siteshwar commented on compose.yaml:
```diff
@@ -37,24 +37,6 @@ services:
     stdin_open: true
     tty: true
 
-  osh-worker:
```
I have temporarily disabled running worker in a container for testing purpose. This would not be in the final pull request.

siteshwar commented on osh/hub/osh_xmlrpc/scan.py:
```diff
@@ -29,6 +40,59 @@
+
+def build_through_resalloc():
+    conn = ResallocConnection(settings.RESALLOC_SERVER)
+    ticket = conn.newTicket(settings.RESALLOC_WORKER_TAGS)
```
@kdudka Where should this ticket id be stored? Shall we create a new field in `Task` table to have a reference to resalloc ticket id?

kdudka replied:
Unless you are implementing a new feature of kobo, the information will need to stored on the OSH level.  Adding new fields (not used by kobo itself) to the `Task` table defined by kobo is difficult.  See https://github.com/release-engineering/kobo/pull/211#issuecomment-1690921525

lzaoral replied:
Sadly, modifying the `Task` model is not possible without breaking projects dependent on `kobo` with do not use or properly apply migrations.

siteshwar replied:
Would it be ok if I store it in the [comment](https://github.com/release-engineering/kobo/blob/8a704d50a8f56de0abfa63f3e383b5df129398fd/kobo/hub/models.py#L561) field of `Task` model? It's a hack, but that makes it easy to do a one to one mapping between task id and [worker events](https://github.com/openscanhub/openscanhub/blob/824beeba205847a66681f95503d7f457f77689de/osh/hub/osh_xmlrpc/worker.py#L39) without introducing a new field or table.

kdudka replied:
Not really.  `comment` is there for users of OpenScanHub.  If they specify a comment with `osh-cli`, they expect to find the exact string in the field later on.

siteshwar commented on osh/hub/osh_xmlrpc/scan.py:
```diff
@@ -29,6 +40,59 @@
+    conn = ResallocConnection(settings.RESALLOC_SERVER)
+    ticket = conn.newTicket(settings.RESALLOC_WORKER_TAGS)
+    # TODO: How long shall we wait here?
+    ip_address = ticket.wait().strip()
```
@praiskup What would happen if this code is running in a container, and the container gets killed?

praiskup replied:
Then nobody closes the ticket, right?  And it stays open forever.  I'd propose having the [resalloc feature](https://github.com/praiskup/resalloc/issues/123) implemented to way-around this.

siteshwar replied:
Shall this be handled in a signal handler that would close the ticket on abnormal exits?

praiskup replied:
Handling KeyboardInterrupt may help, at least for "controlled" shutdowns. But one day you'll have to think about removing leftovers anyway, I'm sure.  It's good to think about this in advance.  Maybe some combination with "self-termination" (configure terminate on shutdown in ec2, and `shutdown -h now` triggered by cron...).

lzaoral committed:
osh: remove all references to GitLab

... and replace them with a corresponding FIXME where applicable.

Related: https://github.com/openscanhub/openscanhub/issues/103
Related: https://gitlab.cee.redhat.com/covscan/covscan/-/issues/196

siteshwar committed:
containers: Cleaning container images should be optional

`prepare_deploy` function should not clean images from previous builds.
Users should explicitly specify `--clean` option if they want to clear
image cache from previous builds.

lzaoral committed:
packaging: fix dependency on `file`

Only worker depends on `file` in osh/worker/csmock_runner.py.

lzaoral committed:
containers: remove superfluous dnf settings

We have already called `dnf install` before this `RUN` command.
Therefore, we have already synced the repositories at least once.

lzaoral committed:
containers: do not install redundant packages

They were either unused or just dependencies of other packages.

lzaoral committed:
containers: do not hard-code dependency on Python 3.6

So that it is easier to switch to newer base image which provides a
different Python version by default.

lzaoral committed:
containers: restructure Dockerfiles to make use of layer caching

Build common parts of Dockerfiles first, so that they can be build only
once, cached and then quickly reused.

lzaoral committed:
containers: add /usr/bin/pg_isready to hub container

... which was removed by mistake.

Fixes the following error:
```
/run.sh: line 9: pg_isready: command not found
/run.sh: line 9: pg_isready: command not found
/run.sh: line 9: pg_isready: command not found
...
```

Fortunately, the error is not fatal and the container would still start
eventually.

Fixes: e309e3e0f674a9bca127128a1f8ebea3a28bae8d ("containers: do not install redundant packages")

lzaoral committed:
hub: fix overlapping pagination controls

Kobo's CSS declare the `.paginator` class as `float`.  Therefore, in
some cases the pagination controls would overlap with other page content
making the visuals look corrupted.

Override the broken style in OSH because changes in kobo's styles may break
other projects that depend on them.

Resolves: https://gitlab.cee.redhat.com/osh/covscan/-/issues/227

lzaoral committed:
hub: remove scan/detail.html template

... because it is unused.

kdudka committed:
packaging: stabilize a new version of osh-client

Closes: https://github.com/openscanhub/openscanhub/pull/130

kdudka committed:
packaging: use gzip to compress distribution tarball

bzip2 is not supported by GitHub when it automatically create snapshots
for a git tag.

Closes: https://github.com/openscanhub/openscanhub/pull/131
